### DISCUSSION TITLE
Can we have different files for each translation (namespace)?

### DISCUSSION THREAD

**Discussion by rawand-faraidun** [amannn/next-intl#620] [Q&A] [answered]
Is it possible to have different JSON files for the translations like this:
![image](https://github.com/amannn/next-intl/assets/62562422/61d84552-805e-45f4-a7a9-867eab6098ea)

I have changed the `messages` folder name to `locales` and our team uses [i18nexus](https://i18nexus.com) for the translations and it adds them by namespaces, I believe that way is easier for later changes

**Comment by amannn** [comment, marked as answer]
Absolutely, see "How can I split my messages into multiple files?" in [the messages configuration docs](https://next-intl-docs.vercel.app/docs/configuration#messages).

**Reply by rawand-faraidun** [reply]
Are there any advantages to using separate files? because as I see in this configuration it will still load all the translations from all files and serves them as one object

**Reply by amannn** [reply]
Personally, I don't use separate files and use [the VSCode integration](https://next-intl-docs.vercel.app/docs/workflows/vscode-integration) to comfortably manage one large file, containing all messages. Some users prefer to have different files to provide more structure though!

It's definitely purely an organizational topic and not something that affects the runtime of your app.

**Reply by rawand-faraidun** [reply]
Thanks, I have recently published a package to help manage translations called [Linguify](https://github.com/rawand-faraidun/linguify), and it works with separate files, guess I have to add support for one file to :)
